src/magic_spell_effect.cpp

```diff
@@ -496,10 +518,10 @@ static void add_effect_to_target( const tripoint &target, const spell &sp, Creat
         for( const bodypart_id &bp : guy->get_all_body_parts() ) {
             if( sp.bp_is_affected( bp.id() ) ) {
                 if( sp.has_flag( spell_flag::LIQUID ) ) {
-                    guy->add_liquid_effect( spell_effect, bp, 1, dur_td, sp.has_flag( spell_flag::PERMANENT ) );
+                    splash_target( target, sp, caster );
```

Comment:
This loop runs for every bp, and the if for every affected bp but splash_target doesn't take bp in argument, won't this risk apllying multiple time the same effect?

Reply:
You're right. splash_target checks the spell's affected_bps and handles it all there, so I can just stick a return at the end of that if and it'll just iterate over the affected BPs once. Regular spells still function as before.